syft/execution/plan.py

```diff
@@ -248,6 +250,40 @@ def add_placeholder(self, tensor, arg_ids, result_ids, node_type=None):
 
         return self.placeholders[tensor.id]
 
+    def add_placeholder_fcall(self, obj, arg_ids, node_type=None):
+        """
```

Comment:
I'm not sure I understand the role of "fcall", what is it here for?

Reply:
I will write better documentation in the code in the next commit. The fcall tag is used to denote the placeholders that were originally in the input. For example if I build a plan with the following arguments:

- th.tensor([1,2])
- [th.tensor([4,5]), th.tensor([6, 7])]
- True

I want to add Placeholders with the #fcall tag (denoting that those placeholders were in the input of the function) and with their types (from build time): FrameworkTensor, list, bool.  The original implementation would not flag the fact that those Placeholders were given as input at build time, they would leave only the #input tag (could not easily deduce what placeholders were used in input function) and the nontensor inputs were not receiving a placeholder, making typing impossible for those specific inputs (nested structures/variables in the future :) ).

Reply:
Not sure I followed everything.
Is the function you're talking about the plan?
Do we have now both #input and #fcall tags? Why having both?
Why did you call it "fcall"?

Reply:
Yes. We want to make plans closer to how actual functions work (without them being actual functions). Let's say we want to build a plan with an underlying function that receives two parameters.

x:  (and the build input) th.tensor([1,2])
y: (and the build input) [th.tensor([4,5]), th.tensor([6, 7])]

The generated placeholders:

- placeholder for th.tensor([1, 2]) will have the following tags: input, fcall, type.
- placeholder for [th.tensor([4, 5]), th.tensor([6, 7])] will have the following tags: fcall, type. (this is not an input tensor)
- placeholder for th.tensor([4, 5]) will have the following tags: input
- placeholder for th.tensor([6, 7]) will have the following tags: input

The fcall tag tells me that this placeholder is a placeholder for a build time argument (even if not FrameworkTensor). This is useful when doing the type check/arguments verification. In the old implementation, the list from the arguments would have not received a Placeholder, making type checking impossible.

The naming might not be the greatest, I'll document this better and think for a better name (suggestions would be great)?